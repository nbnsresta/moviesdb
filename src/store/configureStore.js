import { configureStore } from "@reduxjs/toolkit";
import createSagaMiddleware from "redux-saga";
import rootReducer from "./slices/rootReducer";
import rootSaga from "./sagas/rootSaga";

const isDevelopment = process.env.NODE_ENV !== "production";

const middlewares = [];
const sagaMiddleware = createSagaMiddleware();
middlewares.push(sagaMiddleware);

if (isDevelopment) {
  const { createLogger } = require("redux-logger");
  middlewares.push(createLogger({ collapsed: true }));
}

const storeConfig = () => {
  const store = configureStore({
    reducer: rootReducer,
    middleware: middlewares,
    devTools: isDevelopment
  });
  sagaMiddleware.run(rootSaga);
  return store;
};

export default storeConfig;
