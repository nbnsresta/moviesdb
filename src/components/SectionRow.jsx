import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import Carousel from "react-multi-carousel";
import MovieItem from './MovieItem';
import { categories } from '../store/constants';
import pagesSlice from '../store/slices/pages.slice';

import "react-multi-carousel/lib/styles.css";
import style from './sectionrow.module.css';

const responsive = {
  superLargeDesktop: {
    breakpoint: { max: 4000, min: 3000 },
    items: 8,
  },
  largeDesktop: {
    breakpoint: { max: 3000, min: 1332 },
    items: 6,
  },
  desktop: {
    breakpoint: { max: 1332, min: 1024 },
    items: 5,
  },
  tablet: {
    breakpoint: { max: 1024, min: 776 },
    items: 4,
  },
  tabletSmall: {
    breakpoint: { max: 776, min: 584 },
    items: 3,
  },
  mobileBig: {
    breakpoint: { max: 584, min: 464 },
    items: 2,
  },
  mobile: {
    breakpoint: { max: 464, min: 396 },
    items: 2,
  },
  mobilesmall: {
    breakpoint: { max: 396, min: 0 },
    items: 1,
  },
};

export default ({ title, category }) => {
  const dispatch = useDispatch();
  const movies = useSelector(state => state.movies[category].movies)
  useEffect(() => {
    dispatch(pagesSlice.actions.getMovies({ category: categories[category], page: 1 }));
  }, [category, dispatch]);

  return (
    <div className={style.sectionRow}>
      <h2 className={style.sectionTitle}>{title}</h2>
      <Carousel
        responsive={responsive}
        itemClass="carousel-item-padding-40-px">
        {movies.map(item => <MovieItem key={item.id} movie={item}/>)}
      </Carousel>
    </div>
  )
}