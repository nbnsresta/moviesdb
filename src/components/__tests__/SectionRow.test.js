import React from "react";
import { shallow, mount, render } from "enzyme";
import { Provider } from "react-redux";
import SectionRow from "../SectionRow";
import configureStore from "../../store/configureStore";

let store;
beforeEach(() => {
  store = configureStore();
});

describe("Render | SectionRow", () => {
  it("component renders properly", () => {
    const wrapper = render(
      <Provider store={store}>
        <SectionRow category="new" title="New" />
      </Provider>
    );
    expect(wrapper.html()).toMatchSnapshot();
  });
});
