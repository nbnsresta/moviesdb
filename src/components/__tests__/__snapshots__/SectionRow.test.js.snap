// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders properly 1`] = `"<div class=\\"sectionRow\\"><h2 class=\\"sectionTitle\\">New</h2><div class=\\"react-multi-carousel-list  \\"><ul class=\\"react-multi-carousel-track \\" style=\\"transition: none; overflow: unset; transform: translate3d(0px,0,0);\\"></ul></div></div>"`;
